File: core/src/com/alta_v2/game/screen/GameScreenFactory.java
package com.alta_v2.game.screen;

import com.alta_v2.mediatorModule.screen.context.ScreenContext;

/**
 * Provides the factory to generate screen.
 */
public interface GameScreenFactory {

    /**
     * Creates the game screen to render all UI components.
     *
     * @param screenContext     - the {@link ScreenContext} instance.
     * @return created {@link GameScreen} instance.
     */
    GameScreen createGameScreen(ScreenContext screenContext);

}

File: core/src/com/alta_v2/game/screen/GameScreenFactory.kt
package com.alta_v2.game.screen

import com.alta_v2.mediator.screen.context.ScreenContext

/**
 * Provides the factory to generate screen.
 */
interface GameScreenFactory {
    /**
     * Creates the game screen to render all UI components.
     *
     * @param screenContext     - the [ScreenContext] instance.
     * @return created [GameScreen] instance.
     */
    fun createGameScreen(screenContext: ScreenContext): GameScreen
}